arches/app/search/components/core_search.py

```diff
@@ -0,0 +1,141 @@
+from arches.app.models.system_settings import settings
+from arches.app.search.components.base import BaseSearchFilter
+from arches.app.search.mappings import RESOURCES_INDEX
+from arches.app.utils.permission_backend import user_is_resource_reviewer
+from datetime import datetime
+import json
+
+details = {
+    "searchcomponentid": "69695d63-6f03-4536-8da9-841b07116381",
+    "name": "Core Search",
+    "icon": "",
+    "modulename": "core_search.py",
+    "classname": "CoreSearchFilter",
+    "type": "core",
+    "componentpath": "views/components/search/core-search",
+    "componentname": "core-search",
+    "sortorder": "0",
+    "enabled": True,
+    "config": {
+        "default": True,
+        "requiredComponents": [
+            {
+                "componentname": "advanced-search",
+                "searchcomponentid": "f0e56205-acb5-475b-9c98-f5e44f1dbd2c",
+            },
+            {
+                "componentname": "map-filter",
+                "searchcomponentid": "09d97fc6-8c83-4319-9cef-3aaa08c3fbec",
+            },
+            {
+                "componentname": "paging-filter",
+                "searchcomponentid": "7aff5819-651c-4390-9b9a-a61221ba52c6",
+            },
+            {
+                "componentname": "provisional-filter",
+                "searchcomponentid": "073406ed-93e5-4b5b-9418-b61c26b3640f",
+            },
+            {
+                "componentname": "related-resources-filter",
+                "searchcomponentid": "59f28272-d1f1-4805-af51-227771739aed",
+            },
+            {
+                "componentname": "resource-type-filter",
+                "searchcomponentid": "f1c46b7d-0132-421b-b1f3-95d67f9b3980",
+            },
+            {
+                "componentname": "saved-searches",
+                "searchcomponentid": "6dc29637-43a1-4fba-adae-8d9956dcd3b9",
+            },
+            {
+                "componentname": "search-export",
+                "searchcomponentid": "9c6a5a9c-a7ec-48d2-8a25-501b55b8eff6",
+            },
+            {
+                "componentname": "search-result-details",
+                "searchcomponentid": "f5986dae-8b01-11ea-b65a-77903936669c",
+            },
+            {
+                "componentname": "sort-results",
+                "searchcomponentid": "6a2fe122-de54-4e44-8e93-b6a0cda7955c",
+            },
+            {
+                "componentname": "term-filter",
+                "searchcomponentid": "1f42f501-ed70-48c5-bae1-6ff7d0d187da",
+            },
+            {
+                "componentname": "time-filter",
+                "searchcomponentid": "7497ed4f-2085-40da-bee5-52076a48bcb1",
+            },
+        ],
+    },
+}
+
+SEARCH_RESULT_PAGES = (
+    int(settings.SEARCH_EXPORT_LIMIT // settings.SEARCH_RESULT_LIMIT) - 1
+)
+
+
+class CoreSearchFilter(BaseSearchFilter):
```

Comment:
We shouldn't think of this class as being a type of `SearchFilter`.  Instead it could inherit from a new class called `BaseSearch` that could be in base.py.